cmd/kindnetd/masq.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"time"
+
+	utildbus "k8s.io/kubernetes/pkg/util/dbus"
+	utiliptables "k8s.io/kubernetes/pkg/util/iptables"
+	utilexec "k8s.io/utils/exec"
+)
+
+// NewIPMasqAgent returns a new IPMasqAgent
+func NewIPMasqAgent(ipv6 bool, noMasqueradeCIDRs []string) (*IPMasqAgent, error) {
+	execer := utilexec.New()
+	dbus := utildbus.New()
+	protocol := utiliptables.ProtocolIpv4
+	if ipv6 {
+		protocol = utiliptables.ProtocolIpv6
+	}
+	masqChain := utiliptables.Chain(masqChainName)
+	iptables := utiliptables.New(execer, dbus, protocol)
+
+	// TODO: validate cidrs
+	return &IPMasqAgent{
+		iptables:          iptables,
+		masqChain:         masqChain,
+		noMasqueradeCIDRs: noMasqueradeCIDRs,
+	}, nil
+}
+
+// IPMasqAgent is based on https://github.com/kubernetes-incubator/ip-masq-agent
+// but collapsed into kindnetd and made ipv6 aware in an opionated and simplified
+// fashion
+type IPMasqAgent struct {
+	iptables          utiliptables.Interface
+	masqChain         utiliptables.Chain
+	noMasqueradeCIDRs []string
+}
+
+// SyncRulesForever syncs ip masquerade rules forever
+func (ma *IPMasqAgent) SyncRulesForever(interval time.Duration) error {
+	for {
+		if err := ma.SyncRules(); err != nil {
+			return err
+		}
+		time.Sleep(interval)
+	}
+}
+
+// name of nat chain for iptables masquerade rules
+const masqChainName = "KIND-MASQ-AGENT"
```

Comment:
this is now fixed instead of a flag (the flag came from the original masq agent)